src/votekit/outcome_model.py

```diff
@@ -0,0 +1,140 @@
+        self.curr_round = curr_round
+        self.elected = elected
+        self.eliminated = eliminated
+        self.remaining = remaining
+        self.profile = profile
+        self.winner_votes = winner_votes
+        self.previous = previous
+
+    class Config:
+        allow_mutation = False
+
+    def get_curr_round(self) -> int:
+        return self.curr_round
+    
+    def get_remaining(self) -> list[str]:
+        return self.remaining
+    
+    def get_elected(self) -> list[str]:
+        return self.elected
+    
+    def get_eliminated(self) -> list[str]:
+        return self.eliminated
+    
+    
+    def get_all_winners(self) -> list[str]: # returns in order of first elected - last elected
+        if self.previous != None:
+            return self.previous.get_all_winners() + self.elected
+        else:
+            return self.elected
+    
+    def get_all_eliminated(self) -> list[str]: #returns in order of last eliminated - first eliminated
+        if self.previous != None:
+            l = self.eliminated.copy()
+            l.reverse()
+            return l +  self.previous.get_all_eliminated() 
+        else:
+            l = self.eliminated.copy()
+            l.reverse()
+            return l
+    def get_rankings(self) -> list[str]:
+        return self.get_all_winners() + self.remaining +  self.get_all_eliminated()
+    
+    def get_profile(self) -> PreferenceProfile:
+        if self.profile != None:
+            return self.profile
+        elif self.previous==None:
+            raise ValueError(f"No profile found")
+        else:
+            return self.previous.get_profile()
+    
+    def get_round_outcome(self,roundNum: int) -> list[str]:
+        if self.curr_round == roundNum:
+            # current round's eliminated and/or elected
+            if self.elected != []:
+                return self.elected
+            else:
+                return self.eliminated
+        else:
+            self.previous.get_round_outcome(roundNum)
+    
+  ###############################################################################################
+    
+    """def add_winners_and_losers(self, winners: set[str], losers: set[str]) -> "Outcome":
+        # example method, feel free to delete if not useful
+        if not winners.issubset(self.remaining) or not losers.issubset(self.remaining):
+            missing = (winners.difference(set(self.remaining)) | (losersdifference(set(self.remaining)))
+            raise ValueError(f"Cannot promote winners, {missing} not in remaining")
+        return Outcome(
+            remaining=set(self.remaining).difference(winners | losers),
+            elected=list(set(self.elected) | winners)
+            eliminated=list(set(self.eliminated) | losers)
+        )
+    """
+    
+
+    
+    # THIS ASSUMES BALLOTS DON'T CHANGE BY ROUND!
```

Comment:
assumptions like this should be documented in the docstring and/or enforced via error checking